dojo.py

```diff
@@ -457,8 +453,7 @@ def print_allocations_func(self,room,print_to_file, room_occupants=None):
                     print_to_file = self.data_dir + print_to_file
                 with open(print_to_file, 'a') as allocations_file:
                     allocations_file.write(text)
-                    allocations_file.close()
-                
+                    allocations_file.close()     
```

Comment:
Since you are using a context manager, 
```python
with open(print_to_file, 'a') as allocations_file:
    ...
```
you don't have to close the file manually like you did here.

~allocations_file.close()~